be/src/exprs/agg/aggregate.h

```diff
@@ -351,49 +365,101 @@ class AggregateFunctionBatchHelper : public AggregateFunctionStateHelper<State>
 
     void update_batch(FunctionContext* ctx, size_t chunk_size, size_t state_offset, const Column** columns,
                       AggDataPtr* states) const override {
-        for (size_t i = 0; i < chunk_size; ++i) {
-            static_cast<const Derived*>(this)->update(ctx, columns, states[i] + state_offset, i);
+        if (static_cast<const Derived*>(this)->support_update_with_rows()) {
```

Comment:
We prefer to implement special methods in specific functions rather than modifying the aggregator framework.

Reply:
Overriding them in bitmap_union achieves the same result. If there are more functions that might use them, have them inherit from AggregateFunctionBatchHelper.